### PR TITLE
[dashboard] Add projects empty state illustration for dark theme

### PR DESCRIPTION
**Pull request by jankeromnes**
Fixes https://github.com/gitpod-io/gitpod/issues/4680

<table><tr><th>Light</th><th>Dark</th></tr>
<tr><td>
Screenshot 2021-07-28 at 10 36 54
</td><td>
Screenshot 2021-07-28 at 10 37 11
</td></tr></table>

### CODE REVIEW

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/gitpod-io/gitpod/pull/4981#" title="Author self-approved">jankeromnes</a>*

Associated issue: *#4680*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[components/dashboard/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/dashboard/OWNERS)~~ [jankeromnes]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by jankeromnes** [comment]
Random network flake:

```
Save error occurred:  could not find : chart mysql not found in https://charts.bitnami.com/bitnami: looks like "https://charts.bitnami.com/bitnami" is not a valid chart repository or cannot be reached: stream error: stream ID 1; INTERNAL_ERROR
```

/werft run

File: components/dashboard/src/projects/Projects.tsx
Comment (gtsiolis) [inline components/dashboard/src/projects/Projects.tsx:11]: **praise**: Patterns in action! 😄

File: components/dashboard/src/projects/Projects.tsx
Comment (gtsiolis) [inline components/dashboard/src/projects/Projects.tsx:71]: **issue**: The illustration does not get updated when you change theme unless you reload the page. This may only be noticable when you change theme in another window or when system option is set in theme preferences. Could this be a small fix we could add in this PR? 😭 

| Light Theme | Dark Theme |
|-|-|
| Screenshot 2021-07-28 at 11 53 41 AM | Screenshot 2021-07-28 at 11 53 53 AM |

**Review by gtsiolis — APPROVED** [review]
Thanks for picking this @jankeromnes! 🌔 

Everything works as expected! Left one minor note, feel free to open a follow up issue for this if this isn't an easy fix. 🏖️

**Comment by roboquat** [comment]
LGTM label has been added.  <details>Git tree hash: 6e366d372e84c06d43232533bf5748499dfeae9c</details>

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/gitpod-io/gitpod/pull/4981#pullrequestreview-716744593" title="Approved">gtsiolis</a>*, *<a href="https://github.com/gitpod-io/gitpod/pull/4981#" title="Author self-approved">jankeromnes</a>*

Associated issue: *#4680*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[components/dashboard/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/dashboard/OWNERS)~~ [jankeromnes]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

File: components/dashboard/src/projects/Projects.tsx
Comment (jankeromnes) [reply components/dashboard/src/projects/Projects.tsx:71]: Hmm, there are a few places where we adjust the theme based on `document.documentElement.classList.contains('dark')`, but switching in a different tab doesn't cause a re-render.

This could be fixed, but the fix probably isn't small (will probably require a state + an effect)

Comment (jankeromnes) [reply components/dashboard/src/projects/Projects.tsx:71]: Aha, there is hope: https://caniuse.com/mutationobserver

**Comment by jankeromnes** [comment]
Will try to implement reacting to theme changes.

/hold

**Comment by gtsiolis** [comment]
> Will try to implement reacting to theme changes.

Thanks @jankeromnes! In any case, feel free to open a follow up issue if this isn't easy to fix. This is a minor UX issue and would affect only a handful of users. 🔅

**Comment by roboquat** [comment]
New changes are detected. LGTM label has been removed.

**Comment by jankeromnes** [comment]
Should adjust to theme changes dynamically now 😁 @gtsiolis please take another look.

File: components/dashboard/src/theme-context.tsx
Comment (gtsiolis) [inline components/dashboard/src/theme-context.tsx:36]: **praise**: `actuallySetIsDark` 😂

**Review by gtsiolis — APPROVED** [review]
Smooth! Thanks for tackling also the dynamic change, @jankeromnes! 🔝 

Works like a charm and we'll probably re-use this in other places like the illustration in https://github.com/gitpod-io/gitpod/pull/4884.

**Comment by roboquat** [comment]
LGTM label has been added.  <details>Git tree hash: f68d434023bfd46ca13537ec82b3f6a8f2f15f5d</details>

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/gitpod-io/gitpod/pull/4981#pullrequestreview-716831583" title="Approved">gtsiolis</a>*, *<a href="https://github.com/gitpod-io/gitpod/pull/4981#" title="Author self-approved">jankeromnes</a>*

Associated issue: *#4680*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[components/dashboard/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/dashboard/OWNERS)~~ [jankeromnes]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by gtsiolis** [comment]
Thanks @jankeromnes! UX looks good! Feel free to merge this or forward to someone else to review frontend in-depth if needed.

File: components/dashboard/src/theme-context.tsx
Comment (AlexTugarev) [inline components/dashboard/src/theme-context.tsx:25]: out of curiosity, why do we need an observer for `documentElement`?

**Comment by AlexTugarev** [comment]
/lgtm

this works nicely :-)

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/gitpod-io/gitpod/pull/4981#issuecomment-888253107" title="LGTM">AlexTugarev</a>*, *<a href="https://github.com/gitpod-io/gitpod/pull/4981#pullrequestreview-716831583" title="Approved">gtsiolis</a>*, *<a href="https://github.com/gitpod-io/gitpod/pull/4981#" title="Author self-approved">jankeromnes</a>*

Associated issue: *#4680*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[components/dashboard/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/dashboard/OWNERS)~~ [AlexTugarev,jankeromnes]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

File: components/dashboard/src/theme-context.tsx
Comment (jankeromnes) [reply components/dashboard/src/theme-context.tsx:25]: That's in order to detect when the `classList` changes (receives `dark` class, or `dark` class is removed). This is the main control for light vs dark, and we want to listen to this in order to re-render components that have a programatic dependency on theme (e.g. to swap out an image).

Notes:
- Most of the time, it's better to not listen to `isDark`, but instead use appropriate CSS classes (e.g. `dark:bg-gray-900`)
- Also, this is different from the "theme" preference (which has 3 values: `light`, `dark` and `system`, and actually sets dark or not-dark based on some separate logic)

**Comment by jankeromnes** [comment]
Super cool! Many thanks for the great reviews. 🙏

Ready for launch! 🚀

/unhold

**Comment by roboquat** [comment]
@gtsiolis: cat image

<details>

In response to [this](https://github.com/gitpod-io/gitpod/pull/4981#issuecomment-888292307):

>/meow


Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository.
</details>

**Comment by roboquat** [comment]
@jankeromnes: Why don't skeletons ride roller coasters? They don't have the stomach for it.

<details>

In response to [this](https://github.com/gitpod-io/gitpod/pull/4981#issuecomment-888305940):

>/joke


Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository.
</details>

**Pull request merged**
